ledger/complete/wal/checkpoint_v6_test.go

```diff
@@ -276,6 +276,52 @@ func TestCheckpointV6IsDeterminstic(t *testing.T) {
 	})
 }
 
+func TestWriteAndReadCheckpointV6LeafEmptyTrie(t *testing.T) {
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries := []*trie.MTrie{trie.NewEmptyMTrie()}
+		fileName := "checkpoint-empty-trie"
+		logger := unittest.Logger()
+		require.NoErrorf(t, StoreCheckpointV6Concurrently(tries, dir, fileName, &logger), "fail to store checkpoint")
+		resultChan, err := OpenAndReadLeafNodesFromCheckpointV6(dir, fileName, &logger)
+		require.NoErrorf(t, err, "fail to read checkpoint %v/%v", dir, fileName)
+		for readResult := range resultChan {
+			// only dummy node returned for empty root
+			require.NoError(t, readResult.Err)
+			require.Nil(t, readResult.LeafNode.Path)
+			require.Nil(t, readResult.LeafNode.Payload)
+			require.Nil(t, readResult.LeafNode.Hash)
+		}
+	})
+}
+
+func TestWriteAndReadCheckpointV6LeafSimpleTrie(t *testing.T) {
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries := createSimpleTrie(t)
+		fileName := "checkpoint"
+		logger := unittest.Logger()
+		require.NoErrorf(t, StoreCheckpointV6Concurrently(tries, dir, fileName, &logger), "fail to store checkpoint")
+		resultChan, err := OpenAndReadLeafNodesFromCheckpointV6(dir, fileName, &logger)
+		require.NoErrorf(t, err, "fail to read checkpoint %v/%v", dir, fileName)
+		for readResult := range resultChan {
+			require.NoError(t, readResult.Err, "no errors in read results")
```

Comment:
we should save the result to a list, and then compare with `tries.AllPayloads()`